### PR TITLE
Refactor with redux to re-use ArticleCard in SingleArticle as well

### PR DESCRIPTION
**Pull request by erikbjoern**
[PT link](https://www.pivotaltracker.com/story/show/173040343)

# This PR contains following changes:
* `ArticleCard` is now featured in `SingleArticle` as well
* `ArticleCard` now takes a `size` prop to specify its size. On the index page its size is 1, on single article its 2. The actual sizes is the same as before, multiplied with the `size` prop.
* Redux store holds `articles: { activeArticle: article }` and is set when an article on index page is clicked
* `SingleArticle` is displayed based on `activeArticle`
* This means that the image in `SingleArticle` doesn't have to wait on the get-show-request to load, since it's already loaded in the active article. So it's instantly displayed when a specific article is clicked, no load time.

- Due to merging updated upstream, the `rootReducer` now contains the `combineReducers`, which means that the different resources are accessed as follows: 
```
state: {
  articles: { activeArticle: {} }
  messages: { signupMessage: "" }
  location: { country: "", city: "" }
}
```

### CODE REVIEW

**Review by designerofthing — APPROVED** [review]

**Comment by kermit-klein** [comment]
this looks good for me

**Review by kermit-klein — APPROVED** [review]

File: src/components/ArticleCard.jsx [hunk 1]
```diff
@@ -1,11 +1,22 @@
 import React from "react";
 import { Grid, Image } from "semantic-ui-react";
 import { Link } from "react-router-dom";
+import { useDispatch } from "react-redux";
+
+const ArticleCard = ({ article, size }) => {
+  const dispatch = useDispatch();
```
Comment (jysmys) [inline src/components/ArticleCard.jsx:7]: Dispatch is already declared in the App.jsx so not really needed here, maybe use props instead.

File: src/components/SingleArticle.jsx [hunk 2]
```diff
@@ -1,52 +1,45 @@
...
-  const [article, setArticle] = useState({});
+  const dispatch = useDispatch();
   const { id } = useParams();
   const { t } = useTranslation();
+  const article = useSelector( state => state.articles.activeArticle );
+
+  const chooseArticle = async () => {
+    try {
+      const response = await axios.get(`/articles/${id}`);
+      dispatch({ type: "SET_ACTIVE_ARTICLE", payload: response.data.article });
```
Comment (jysmys) [inline src/components/SingleArticle.jsx:22]: use props.dispatch here ? then the import is not necessary...

File: src/state/store/configureStore.js [hunk 3]
```diff
@@ -1,6 +1,14 @@
-import { createStore } from "redux";
+import { createStore, compose } from "redux";
 import rootReducer from "../reducers/rootReducer";
+
 const configureStore = () => {
-  return createStore(rootReducer);
+  return createStore(
+    rootReducer,
+    compose(
+      window.__REDUX_DEVTOOLS_EXTENSION__ &&
```
Comment (jysmys) [inline src/state/store/configureStore.js:8]: what does this do?

**Pull request merged**